src/server/types.ts

```diff
@@ -408,3 +412,9 @@ export interface PluginRoute {
   // deno-lint-ignore no-explicit-any
   handler?: Handler<any, any> | Handlers<any, any>;
 }
+
+export interface PluginIsland {
```

Comment:
Are name and path necessary? In my perfect world I see something like this:

`
{
  islands: [
    IslandComponent1,
    IslandComponent2
  ]
}
`

Is the location field in the Plugin type also necessary?

Reply:
Currently the location is necessary to be able to feed it into esbuild. We could explore creating a virtual entry point for these though.

Reply:
Ok, I understand :-) Maybe it would be better to transform the islands field to an object:

```typescript
{
  ...
  islands: {
    location: '...',
    entries: [
      {
        name: '...',
        path: '...',
        component: MyIslandComponent
      }
    ]
  }
  ...
}
```

Or something similar. With this model it is more clear that location belongs to the islands.

Reply:
@lino-levan How do you bundle those islands for the browser when all you have is a reference to the function provided by the plugin? That's something I'm having trouble figuring out.

Reply:
I see the issue. Definitely non-trivial. 

This feels pretty hacky but we could just
```tsx
function test() {
  // contents
}

export const plugin = {
    islands: [test]
}
```
and then just do
```tsx
test.toString()
```
on the other side to extract the actual function definition.

Reply:
The `island.toString()` only works when it's pure and not accessing variables or identifiers outside of the function body like import statements etc. When I look at the code bases here at Deno, most islands are not pure and access identifiers outside of the function body.

Reply:
Ah you're right. I don't see a way out of this one. Is there any reason we're even specifying the component at all? It seems like we need to bundle it regardless?

Reply:
What do we think about this?
```tsx
    islands: {
      location: import.meta.url,
      names: ["./sample_islands/IslandFromPlugin.tsx"],
    },
```

Reply:
It is fine for me. I would rename the name field to paths and the location field to baseLocation. That is for me a bit clearer but it is only my opinion.

Many thanks for your work. With this approach we can create cool plugins now and that is also a first good approach to structure big Fresh applications better.

Reply:
Seems like a reasonable suggestion. I renamed both, given that there were no other comments on the topic.